api/client/request.go

```diff
@@ -210,31 +210,14 @@ func headerVal(key string, resp *http.Response) (int, bool) {
 
 func parseHTTPStatus(resp *http.Response, body []byte) error {
 
-	var (
-		status *Status
-		err    error
-	)
-
 	httpOK := resp.StatusCode >= http.StatusOK && resp.StatusCode <= http.StatusPartialContent
-	hasStatus := false
-	if body != nil {
-		err = json.Unmarshal(body, status)
```

Comment:
This line first of all was wrong.
It should have been `json.Unmarshal(body, &status)`

Secondly regardless of the response from the server , the status instance was always nil [before above correction] or empty instance[ after correction]

After careful observation , it turns out its not needed at all.
Since the status code is enough to tell us if the body contains a json string for the client to unmarshal or an error string.

Reply:
status is a pointer, do we need to pass &status?  You're relying on the http error to be returned on error, this may not be true. You need to verify that is before this change goes in.